test/scenarios.js

```diff
@@ -949,4 +933,320 @@ describe('Scenarios', async () => {
+  }).timeout(5000);
+
+  it('BlockReward changes both before or after confirming block, check for randao penalty', async () => {
+    // Randao Penalty should be applied correctly when
+    let epoch = await getEpoch();
+    let votes = await getVote(medians);
+    const commitment = utils.solidityKeccak256(
+      ['uint32', 'uint48[]', 'bytes32'],
+      [epoch, votes, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+    );
+    await voteManager.connect(signers[1]).commit(epoch, commitment);
+
+    await mineToNextState(); // reveal
+
+    await voteManager.connect(signers[1]).reveal(epoch, votes,
+      '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+
+    await mineToNextState(); // propose
+
+    const stakerId = await stakeManager.stakerIds(signers[1].address);
+    const staker = await stakeManager.getStaker(stakerId);
+
+    const { biggestInfluence, biggestInfluencerId } = await getBiggestInfluenceAndId(stakeManager, voteManager);
+    const iteration = await getIteration(voteManager, stakeManager, staker, biggestInfluence);
+    await blockManager.connect(signers[1]).propose(epoch,
+      medians,
+      iteration,
+      biggestInfluencerId);
+
+    await mineToNextState(); // dispute
+    await mineToNextState(); // confirm
+    let stake = await stakeManager.getStake(stakerId);
+    await blockManager.connect(signers[1]).claimBlockReward();
+
+    stake = await stakeManager.getStake(stakerId);
+    const updateBlockRewardTo = stake.sub(tokenAmount('100'));
+    await governance.connect(signers[0]).setBlockReward(updateBlockRewardTo);
+
+    await mineToNextEpoch();
+    epoch = await getEpoch();
+
+    votes = await getVote(medians);
+
+    const commitment2 = utils.solidityKeccak256(
+      ['uint32', 'uint48[]', 'bytes32'],
+      [epoch, votes, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+    );
+    await voteManager.connect(signers[1]).commit(epoch, commitment2);
+
+    await mineToNextState(); // didn't reveal
+    await mineToNextState(); // didn't propose
+    await mineToNextState(); // didn't dispute
+    await mineToNextState(); // didn't confirmed
+
+    await mineToNextEpoch();
+    epoch = await getEpoch();
+    const stakeBefore = await stakeManager.getStake(stakerId);
+
+    votes = await getVote(medians);
+    const commitment3 = utils.solidityKeccak256(
+      ['uint32', 'uint48[]', 'bytes32'],
+      [epoch, votes, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+    );
+    await voteManager.connect(signers[1]).commit(epoch, commitment3);
+    stake = await stakeManager.getStake(stakerId);
+
+    assertBNEqual(stake, stakeBefore.sub(updateBlockRewardTo), 'Randao Penalty is not being applied correctly');
+  }).timeout(5000);
+
+  it('Staker will not be able to reveal if minstake increases to nowStake of staker during reveal state', async () => {
```

Comment:
what does nowStake mean?

Reply:
Hey it was just for explanation in testing channel forgot to change it can be written as currentStake/ presentStake or any other synonym. 
Made it currentStake though.